### DISCUSSION TITLE
SSSD federation

### DISCUSSION THREAD

**Discussion by Keeper-of-the-Keys** [keycloak/keycloak#15587] [Q&A]
The keycloak [documentation](https://www.keycloak.org/docs/latest/server_admin/#sssd-and-d-bus) mentions running a script called federation-sssd-setup.sh to use federation via SSSD (which allows using FreeIPA OTP as far as I understand), I can't find this script anywhere what package is supposed to provide this? Does the documentation need to be updated?

**Comment by abstractj** [comment]
@Keeper-of-the-Keys the `federation-sssd-setup.sh` script was removed on https://github.com/keycloak/keycloak/commit/390c7485c78db6face9509e2f2e03ec118917f06, but it seems that we missed the updates into our docs.

@stianst do you think we should move that script to another place? Or add it as part of the docs?

**Comment by Keeper-of-the-Keys** [comment]
Cool, from looking at the diff it looks like the script just does some basic settings in sssd.conf and PAM, do these steps still need to be taken? And how do I cause SSSD to become an option inside of the keycloak UI itself?

**Comment by abstractj** [comment]
@Keeper-of-the-Keys it's been a while since I touched, so please take my word with a grain of salt. The script previously provided are just regular configuration steps, we used to ship our release to make it easy for people, considering how complicated was the whole setup.
After all the configuration steps, in theory the Keycloak server should be able to identify that you have SSSD up and running, properly configured, and the federation provider should be available.

**Comment by Keeper-of-the-Keys** [comment]
I made the changes and I'm pretty sure I also implemented all other things listed in the documenation (install jna, libunix-dbus-java and ran sssctl successfully) but I still have no option to add an SSSD provider :/ 
Any suggestions (20.0.1)?

**Comment by Keeper-of-the-Keys** [comment]
Was SSSD support removed from 20.x.y (20.0.1)? @abstractj @stianst

**Comment by Keeper-of-the-Keys** [comment]
- The commit referred above seems to remove sssd support as well as the setup script.
- I installed 19.0.3 there SSSD is an option (but fails for a reason I have not yet succeeded in debugging)

**Comment by s1341** [comment]
I also need the SSSD support. Is there any chance it will be restored?

**Reply by s1341** [reply]
Opened an issue https://github.com/keycloak/keycloak/issues/16165

**Comment by Keeper-of-the-Keys** [comment]
@s1341 Since I was just playing around with different solutions and not pinned down to something in Keycloak I ended up switching to the recently revived Ipsilon, I also still hope to try authentik at some point though that I believe also does not do SSSD (and thus no TOTP from freeIPA).

**Reply by s1341** [reply]
Hrm. Does ipsilon support openid and 2fa (web-authn)?

**Comment by Keeper-of-the-Keys** [comment]
As far as I understand...
I haven't as yet tied it to a different system and have other priorities...
https://ipsilon-project.org/

Op zo 25 dec. 2022 om 11:18 schreef s1341 ***@***.***>:
